### DISCUSSION TITLE
sandbox methods organization

### DISCUSSION THREAD

**Discussion by torohaifisch** [apollographql/apollo-server#6246] [Q&A] [answered]
Hello i dont know if this question goes here but, my question is related to the sandbox and federation.

When you have many federated services connected in a gateway, the sandbox gets super unorganized with a lot of querys and mutations. Is there a way to split or classify the requests per service or custom sections? or if this isnt possible what would you use to create the graphql documentation in a better and organized way.

**Comment by papollosc** [comment]
Hi @torohaifisch Thank you for your query. If we understand your use case correctly, I think our most recent release to Explorer should be able to help. Check it out and please let me know if that makes organization of querys and mutations easier for you. https://www.apollographql.com/blog/announcement/platform/save-and-share-your-graphql-operations-in-apollo-explorer/

**Comment by torohaifisch** [comment]
hi @papollosc thanks for the answer, yes this is sort of what i need but is missing some features i would like. An example is like in postman you can create collections and inside those you can create folders to better organize your requests, i think that would be a nice addition. 

From what i understood from reading the announcements and doing some testing is that when you use the local sandbox it shows all your local collections, it doesnt matter if they arent related to the graph you are connected to. This doesnt happen if you use apollo studio i think?. Correct me if i am wrong but when you are using a specific graph in the local sandbox you can still see collections not related to the graph or even to the project you are working on so maybe it would be cool to have something like postman workspaces so you can have collections organized by project or something like that.

**Comment by papollosc** [comment, marked as answer]
> when you are using a specific graph in the local sandbox you can still see collections not related to the graph

Thank you for this feedback @torohaifisch and very sorry for the delay in getting back to you. Clarification on the point above, are you pulling registered graphs from Studio? or pointing the endpoint in Sandbox to different graphs? 

Yes, if you have graphs registered in Studio, you can create collections within the scope of a graph/variant only. 
In Sandbox, you can also pull collections of the registered graph/variants. But currently, collections created in Sandbox _for_ Sandbox are across all endpoints that Sandbox points to.

**Reply by torohaifisch** [reply]
hi, yes i was pointing the sandbox to different endpoints, like to endpoint 1 and created a collection then switched to endpoint 2 and i could still see the collection made with the methods of endpoint 1. For me this doesnt make sense as there is no point in seeing a collection that doesnt work with the current pointed endpoint. That is why i suggested something like workspaces to split the collections per endpoint or something like that.

**Reply by cheapsteak** [reply]
Hi there

We eventually would like to reorganize our information architecture so that sandboxes are scoped to a specific graph variant that would hopefully solve for this case. It would be as if each graph variant were a workspace. Hope that once that's available it will help ameliorate the problem with seeing operations for unrelated graphs :)

**Reply by torohaifisch** [reply]
also it would be cool on the federated sandbox to identify the operations by the graph from where they originate
